frame/vamm/src/helpers/math.rs

```diff
@@ -25,11 +29,9 @@ impl<T: Config> Pallet<T> {
 		ensure!(!base.is_zero(), Error::<T>::BaseAssetReserveIsZero);
 		ensure!(!quote.is_zero(), Error::<T>::QuoteAssetReserveIsZero);
 
-		let base_u256 = Self::balance_to_u256(base)?;
-		let quote_u256 = Self::balance_to_u256(quote)?;
-		let invariant = base_u256
-			.checked_mul(quote_u256)
-			.ok_or(Error::<T>::FailedToDeriveInvariantFromBaseAndQuoteAsset)?;
+		let base_u256 = base.into_u256();
+		let quote_u256 = quote.into_u256();
+		let invariant = base_u256.checked_mul(quote_u256).ok_or(Overflow)?;
```

Comment:
Doesn't `.try_mul()` work in this case?

Reply:
`.try_mul()` requires `UnsignedMath`, which is not implemented to `U256` yet.
(I'd like to implement it and then have a better way to work with `U256` values, but I'm still lacking the knowledge to do so. Gotta study more about Rust :books: )

Reply:
If you don't own `UnsignedMath` or `U256` you won't be able to implement `UnsignedMath` for `U256`

Reply:
We do own `UnsignedMath` (`frame/composable-maths/src/labs/numbers.rs:45`), but the requirements for it are not satisfied by `U256`. I have to study more about how to implement it :sweat_smile: